sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -275,7 +275,7 @@ object QueryExecution {
     Seq(
       // `AdaptiveSparkPlanExec` is a leaf node. If inserted, all the following rules will be no-op
       // as the original plan is hidden behind `AdaptiveSparkPlanExec`.
-      InsertAdaptiveSparkPlan(sparkSession),
+      InsertAdaptiveSparkPlan(AdaptiveExecutionContext(sparkSession)),
```

Comment:
It seems clearer to me to create `AdaptiveExecutionContext` inside `InsertAdaptiveSparkPlan` rule, and just pass `SparkSession` to the rule.